layer/layer.go

```diff
@@ -188,6 +188,8 @@ type Store interface {
 	GetRWLayer(id string) (RWLayer, error)
 	GetMountID(id string) (string, error)
 	ReleaseRWLayer(RWLayer) ([]Metadata, error)
+	LockRWLayer()
+	UnlockRWLayer()
```

Comment:
@aaronlehmann I've tried to lock the rw layer just in the progress of `TarStream` but found it was not enough. In other words, #29654 was still not resolved.

I found that the dm device would be busy as long as it is mounted. That's why I added these two interfaces (I knew it was a bit ugly though).